codeworld-server/src/Main.hs

```diff
@@ -354,6 +358,31 @@ runHandler = do
     modifyResponse $ setContentType "text/javascript"
     serveFile (buildRootDir mode </> targetFile programId)
 
+exportAndroidHandler :: Snap()
+exportAndroidHandler = do
+    mode <- getBuildMode
+    Just source <- getParam "source"
+    maybeAppName <- getParam "appName"
+    let appName = BC.unpack $ fromMaybe (BC.pack "CodeWorld App") maybeAppName 
+    let programId = sourceToProgramId source
+        deployId = sourceToDeployId source
+    success <- liftIO $ do
+        ensureProgramDir mode programId
+        B.writeFile (buildRootDir mode </> sourceFile programId) source
+        writeDeployLink mode deployId programId
+        compileIfNeeded mode programId
+    unless success $ modifyResponse $ setResponseCode 500
+    let appProps = M.fromList [("appName", appName)]
```

Comment:
This AppProps thing is entirely unnecessary.  You're trying to invent a dynamically typed language in Haskell, and it's ugly.  Just go ahead and commit to defining a data type with the right fields.  (Which, at the moment, is just one String, right?)